internal/wasi/wasi.go

```diff
@@ -1558,25 +1569,29 @@ type Config struct {
 	// environ stores each environment variable in the form of "key=value",
 	// which is both convenient for the implementation of environ_get and matches os.Environ
 	environ *nullTerminatedStrings
-	stdin   io.Reader
-	stdout,
-	stderr io.Writer
-	opened map[uint32]fileEntry
+	opened  map[uint32]fileEntry
 	// timeNowUnixNano is mutable for testing
 	timeNowUnixNano func() uint64
 	randSource      func([]byte) error
 }
 
+// FD number constants of standard input / output / error.
+const (
+	stdinFD = iota
+	stdoutFD
+	stderrFD
+)
+
 func (c *Config) Stdin(reader io.Reader) {
-	c.stdin = reader
```

Comment:
not currently planning to do anything to try to make stdin/stdout/stderr act like normal files as they aren't and it is more distracting forcing them though a struct that only has one side useful than the special casing in the first place.